File: server/src/motor/motor_service.py
from mqtt.client import MQTTClient
from motor.motor import Motor

class MotorService:
    def __init__(self, motor: Motor):
        self.__motor = motor;

    def move_motor(self, n_steps: int):
        client = MQTTClient.get_instance()
        client.connect('127.0.0.1')
        client.publish('/motormove', n_steps)

File: client/src/services/lib/motor-service.ts
import { MQTTClient } from "../../mqtt";

export class MotorService {
  private deviceId: string;

  public constructor(deviceId: string) {
    this.deviceId = deviceId;
  }

  public async move(numberOfSteps: number): Promise<void> {
    const client = MQTTClient.getInstance();
    client.publish(`device/${this.deviceId}/move`, numberOfSteps.toString());
  }
}
